File: leave-service/src/main/java/com/leave/management/leave/service/LeaveRequestService.java
package com.leave.management.leave.service;

import com.leave.management.leave.dto.LeaveApprovalRequest;
import com.leave.management.leave.dto.LeaveRequestCreateRequest;
import com.leave.management.leave.dto.LeaveRequestResponse;

import java.util.List;

public interface LeaveRequestService 
{
    LeaveRequestResponse applyLeave(Long employeeId,LeaveRequestCreateRequest request);

    LeaveRequestResponse getLeaveRequestById(Long id);

    List<LeaveRequestResponse> getEmployeeLeaveHistory(Long employeeId);

    List<LeaveRequestResponse> getManagerPendingRequests(Long managerId);

    LeaveRequestResponse processLeaveRequest(Long requestId,Long managerId,LeaveApprovalRequest request);
}

File: frontend/src/services/leaveRequestService.js
import axiosInstance from "./api/axiosInstance";
import API_CONFIG from "./api/apiConfig";

const leaveRequestService = {
    createLeaveRequest: async (requestData) => {
        const response = await axiosInstance.post(
            API_CONFIG.endpoints.leaveRequests.base,
            requestData
        );

        return response.data;
    },

    applyLeave: async (employeeId, requestData) => {
        const payload = {
            ...requestData,
            employeeId,
        };

        return leaveRequestService.createLeaveRequest(payload);
    },

    getLeaveRequestById: async (id) => {
        const response = await axiosInstance.get(
            API_CONFIG.endpoints.leaveRequests.byId(id)
        );

        return response.data;
    },

    getEmployeeLeaveHistory: async (employeeId) => {
        const response = await axiosInstance.get(
            API_CONFIG.endpoints.leaveRequests.byEmployee(
                employeeId
            )
        );

        return response.data;
    },

    getManagerPendingRequests: async (managerId) => {
        const response = await axiosInstance.get(
            API_CONFIG.endpoints.leaveRequests.pendingByManager(
                managerId
            )
        );

        return response.data;
    },

    processLeaveRequest: async (requestId, approvalData) => {
        const response = await axiosInstance.patch(
            API_CONFIG.endpoints.leaveRequests.process(
                requestId
            ),
            approvalData
        );

        return response.data;
    },
};

export default leaveRequestService;